entity-services/src/main/xdmp/entity-services/entity-services-impl.xqy

```diff
@@ -158,7 +158,7 @@ declare private variable $esi:model-schematron :=
          <iso:assert test="not( matches( string(.), '^es|json|xsi|xs|xsd|[xX][mM][lL]$' ) )">Namespace prefix <xsl:value-of select="." /> is not valid.  It is a reserved pattern.</iso:assert>
         </iso:rule>
         <iso:rule context="/">
-         <iso:assert test="count(.//(es:namespace-prefix|namespacePrefix)) eq count(distinct-values(.//(es:namespace-prefix|namespacePrefix)))">You cannot duplicate a namespace prefix within a model descriptor.</iso:assert>
+         <iso:assert test="count(distinct-values(.//(es:namespace|namespace))) eq count(distinct-values(.//(es:namespace-prefix|namespacePrefix)))">You cannot duplicate a namespace prefix within a model descriptor.</iso:assert>
```

Comment:
Can we edit the message so it applies to both namespace and ns prefix. Maybe something like:
"You cannot duplicate a namespace or a prefix within a model descriptor."

Reply:
hm, you're right that the message is bad... also that test, it occurs to me, it bad :(

This is a false pass:

a:http://xxx
b:http://xxx
b:http://yyy

because the distinct values of both prefixes and uris is 2!

i have to get a little more clever here.

Reply:
How about asserting them separately within the context "/"

```
<iso:assert test="count(.//(es:namespace-prefix|namespacePrefix)) eq count(distinct-values(.//(es:namespace-prefix|namespacePrefix)))">You cannot duplicate a namespace prefix within a model descriptor.</iso:assert>
<iso:assert test="count(.//(es:namespace|namespace))) eq count(distinct-values(.//(es:namespace-prefix|namespacePrefix)))">You cannot duplicate a namespace within a model descriptor.</iso:assert>
```

Reply:
my comment wont work when all types have same uri and prefix. I will stick to testing.

Another false negative:
When a type has either no uri or namespace, your code will err out with msg `"duplicate namespace prefix"`  because of count mismatch alongwith `"If a type has a namespace property, it must also have a namespacePrefix property"`

Reply:
what we have to do is count distinct prefixes, uris, and the two concatenated together.

Reply:
I have my tests ready. Once there is an update to messaging and assertion logic, I will update my expected result and create a PR.